integration/kafka/src/main/java/alluxio/kafka/connect/AlluxioSinkTask.java

```diff
@@ -0,0 +1,177 @@
+  @Override
+  public void start(Map<String, String> map) {
+    mConfig = new AlluxioSinkConnectorConfig(map);
+    String formatClassName = mConfig.getString(AlluxioSinkConnectorConfig.ALLUXIO_FORMAT);
+    try {
+      mFormat = (AlluxioFormat) Class.forName(formatClassName).newInstance();
+    } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to initiate class" + formatClassName);
+    }
+
+    String alluxioUrl = mConfig.getString(AlluxioSinkConnectorConfig.ALLUXIO_URL);
+    String topDir = mConfig.getString(AlluxioSinkConnectorConfig.TOPICS_DIR);
+    String topicsUrl = alluxioUrl + "/" + topDir;
+    String tmpTopicsUrl = topicsUrl + "/tmp";
+
+    //create topics dir
+    AlluxioURI topicsDirPath = new AlluxioURI(topicsUrl);
+    try {
+      if (!mFs.exists(topicsDirPath)) {
+        mFs.createDirectory(topicsDirPath);
+      }
+      AlluxioURI tmpTopicsDirPath = new AlluxioURI(tmpTopicsUrl);
+      if (!mFs.exists(tmpTopicsDirPath)) {
+        mFs.createDirectory(tmpTopicsDirPath);
+      }
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to create " + topDir + " dir");
+    }
+
+    Collection<TopicPartition> assignments = context.assignment();
+    try {
+      initializeTopicPartitionWriter(assignments);
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception is {}", e);
+      throw new RuntimeException("Failed to initialize TopicPartitionWriter");
+    }
+  }
+
+  /**
+   * Initialize AlluxioTopicPartitionWriter according to task assignments.
+   * One TopicPartition corresponds to one stream.
+   *
+   * @param assignments assigned TopicPartitions for this task
+   */
+  private void initializeTopicPartitionWriter(Collection<TopicPartition> assignments)
+    throws AlluxioException, IOException {
+    for (TopicPartition tp : assignments) {
+      AlluxioTopicPartitionWriter tpWriter =
+          new AlluxioTopicPartitionWriter(mConfig, context, mFormat, mFs, tp);
+      tpWriter.initialize();
+      mTpWriters.put(tp, tpWriter);
+    }
+  }
+
+  @Override
+  public void put(Collection<SinkRecord> records) {
+    for (SinkRecord record : records) {
+      String topic = record.topic();
+      int partition = record.kafkaPartition();
+      TopicPartition topicPartition = new TopicPartition(topic, partition);
+      AlluxioTopicPartitionWriter writer = mTpWriters.get(topicPartition);
+      if (writer != null) {
+        writer.writeToBuffer(record);
+      } else {
+        LOG.error("Failed to get writer of " + topicPartition.toString());
+      }
+    }
+
+    for (AlluxioTopicPartitionWriter writer : mTpWriters.values()) {
+      try {
+        writer.writeRecord();
+      } catch (IOException | AlluxioException e) {
+        LOG.error("Failed to write record to " + writer.getTopicPartition().toString());
```

Comment:
In case of exception, do we just throw the SinkRecords? This can potentially lose data. We need a mechanism to handle transient exceptions and fail the connector in case of the